scripts/provision_credentials.py

```diff
@@ -0,0 +1,141 @@
+    return json.dumps({'key': key, 'key_id': str(uuid.uuid4())})
+
+
+def create_aws_secrets_manager_secret(client, secret_name, value_callback):
+    """
+    Only creates a secret if one with the same name does not currently exist.
+    Otherwise nothing is done (even if value is different from what's currently stored)
+
+    We want value as a callback (that takes no args) because we want to get the value
+    lazily in case we don't end up creating the secret
+    """
+    qualified_name = config.secrets_manager_secret_name(secret_name)
+    # check if the secret exists first for lazy evaluation of value_callback
+    try:
+        client.describe_secret(SecretId=qualified_name)
+    except ClientError as e:
+        if e.response['Error']['Code'] != 'ResourceNotFoundException':
+            raise
+    else:
+        logger.info('AWS secret %s already exists. No changes will be made.', qualified_name)
+        return
+    try:
+        response = client.create_secret(Name=qualified_name, SecretString=value_callback())
+    except ClientError as e:
+        if e.response['Error']['Code'] == 'ResourceExistsException':
+            logger.info('AWS secret %s already exists. No changes will be made.', qualified_name)
+            # FIXME: could add a cleanup callback
+            logger.warning('Resources may have been allocated for secret, but not stored')
+        else:
+            raise
+    else:
+        assert response['Name'] == qualified_name
+        logger.info('Successfully created new AWS secret %s.', qualified_name)
+
+
+def destroy_aws_secrets_manager_secret(client, secret_name):
+    qualified_name = config.secrets_manager_secret_name(secret_name)
+    try:
+        response = client.delete_secret(SecretId=qualified_name,
+                                        ForceDeleteWithoutRecovery=True)
+    except ClientError as e:
+        if e.response['Error']['Code'] == 'ResourceNotFoundException':
+            logger.info('AWS secret %s doesn\'t exist. No changes will be made.', qualified_name)
+        else:
+            raise
+    else:
+        assert response['Name'] == qualified_name
+        logger.info('Successfully deleted AWS secret %s.', qualified_name)
+
+
+def provision_credentials(account_email):
+    client = boto3.client('secretsmanager')
+    # Google service account
+    creds = partial(create_service_account_creds, account_email)
+    create_aws_secrets_manager_secret(client, 'google_service_account', creds)
+    # HMAC key
+    create_aws_secrets_manager_secret(client, 'hmac', partial(random_hmac_key))
+
+
+def destroy_credentials(account_email):
+    client = boto3.client('secretsmanager')
+    destroy_service_account_creds(client,  account_email)
+    destroy_aws_secrets_manager_secret(client, 'google_service_account')
+    destroy_aws_secrets_manager_secret(client, 'hmac')
+
+
+if __name__ == "__main__":
+    action = sys.argv[1]
```

Comment:
Consider spending 10min to add an argparse parser with help string etc.

Reply:
Argparser made.